src/java.desktop/macosx/classes/com/apple/laf/AquaComboBoxPopup.java

```diff
@@ -185,8 +185,14 @@ protected Rectangle adjustPopupAndGetBounds() {
             updateContents(true);
         }
 
+        int popupBoundsY = comboBox.getBounds().height;
+        if (comboBox.isEditable() && comboBox.getBorder() != null) {
+            Insets inset = comboBox.getBorder().getBorderInsets(comboBox);
+            popupBoundsY += inset.top + inset.bottom;
+        }
+
```

Comment:
Did we check where the popup appears for the native components?

Reply:
It does pop up on native components now that it's checked. I just tried applying the same fix for non-editable comboBoxes but the position doesn't change. Seems to be something overriding the position for non-editable comboBoxes. Looking into this

Reply:
For editable comboboxes, the popup is placed below the combobox 
but for non-editable comboboxes, the popup is placed obscuring the selected item showing the list with the selected item  being highlighted
This can be seen in SystemPreference->Date&Time(for editable) and "Open Language&region"(for non-editable) so I think we should only fix it for editable one and not do anything for non-editable one..

Reply:
After some digging, I agree with Prasanta. The native example given works as described. So for non-editable comboboxes, the popup should mostly overlap the combobox. However, there was still a popup location difference between a non-editable combobox with a border, so I included that fix as well.

In summary, the popup now matches the native components Prasanta referenced above.